src/test/scala/fi/vm/sade/hakurekisteri/integration/valintatulos/ValintaTulosActorWithRedisSpec.scala

```diff
@@ -215,6 +220,38 @@ class ValintaTulosActorWithRedisSpec extends ScalatraFunSuite with FutureWaiting
     )
   }
 
+  test("ValintaTulosActor should answer correctly also during cache update") {
+    withSystem(
+      implicit system => {
+        implicit val formats: Formats = HakurekisteriJsonSupport.format
+        implicit val ec: ExecutionContextExecutorService = ExecutorUtil.createExecutor(10, classOf[ValintaTulosActorWithRedisSpec].getSimpleName)
+        val endPoint = createEndPoint
+        val valintaTulosActor = system.actorOf(Props(new ValintaTulosActor(config = config, cacheFactory = cacheFactory, client = new VirkailijaRestClient(config = vtsConfig, aClient = Some(new CapturingAsyncHttpClient(endPoint))))))
+
+        val hakuOid = "1.2.246.562.29.70576649506"
+        when(endPoint.request(forPattern("http://localhost/valinta-tulos-service/haku/" + hakuOid))).thenAnswer(new Answer[(Int, List[(String, String)], String)] {
+          override def answer(invocation: InvocationOnMock): (Int, List[(String, String)], String) = {
+            //Thread.sleep(100)
```

Comment:
Nämä kommentoidut voisi varmaan periaatteessa siivoa pois – toisaalta niistä voi olla hyötyä näitä juttuja debuggaillessa myöhemmin :)